### PR TITLE
Tb/xsx pipeline

### PR DESCRIPTION
**Pull request by toddbellMSFT**
contains pipelines for Unreal to work on our GDK machine.

ALSO contains pipelines that haven't been checked in yet for vanilla Unreal on Windows.

### CODE REVIEW

**Review by Vicodex — APPROVED** [review]

File: UnrealGdkResourceTemplate.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      name: PlayFab/SdkGenerator
+    - repository: UnrealMarketplacePlugin
+      endpoint: GitHub_PlayFab
+      type: github
+      name: PlayFab/UnrealMarketplacePlugin
+
+jobs:
+- job: UnrealGdkResourceTemplate
+  steps:
+  - bash: echo UnrealGdkResourceTemplate
```
Comment (pgilmorepf) [inline UnrealGdkResourceTemplate.yml:29]: I think now that we have a working pattern in place, we don't need these echo-debug jobs. The "cost" of this is pretty high (it has to instantiate an entire job just to print one line)

File: UnrealGdkJobTemplate.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# JOB LEVEL TEMPLATE:
```
Comment (pgilmorepf) [inline UnrealGdkJobTemplate.yml:1]: Template files should not be in this repo.
They should be here: https://dev.azure.com/PlayFabInternal/Main/_git/UnrealGdkTemplate

File: UnrealGdkStepTemplate.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    uePath="C:/dev/UE4_GDK_PFP_MAIN/Engine/Build/BatchFiles"
+
+    exampleProjectPath=$WORKSPACE/sdks/$SdkName/$ueVersion/ExampleProject/ExampleProject.uproject
+
+    targetPlatform=WinGDK
+
+    archivePath=Build/$targetPlatform
+
+    pushd $uePath
+    ./RunUAT.bat BuildCookRun -rocket -nop4 -project="$WORKSPACE/sdks/$SdkName/$ueVersion/ExampleProject/ExampleProject.uproject" -cook -stage -archive -archivedirectory="$archivePath" -package -clientconfig=Development -Platform=$targetPlatform -compressed -CookAll -pak -prereqs -build -utf8output -editorrecompile
```
Comment (pgilmorepf) [inline UnrealGdkStepTemplate.yml:89]: I like putting the meaty statements into a .sh file, as functions, and then the yml bash is just a series of function calls.
https://dev.azure.com/PlayFabInternal/Main/_git/UnityGdkTemplate?path=%2FunityGdk.sh

**Review by pgilmorepf — CHANGES_REQUESTED** [review]
The biggest problem in my mind is that these files shouldn't be here. They should be here:
https://dev.azure.com/PlayFabInternal/Main/_git/UnrealGdkTemplate
This whole PR is private stuff that should not be customer visible, and you should move it to the template first, and then purge this PR branch, so that it's clean and even its history doesn't contain these files.

**Comment by pgilmorepf** [comment]
After you move these files to the internal template location, I'll review more deeply.

**Pull request closed without merging**